de.dlr.sc.virsat.model.edit/src/de/dlr/sc/virsat/model/concept/types/structural/ABeanStructuralElementInstance.java

```diff
@@ -109,6 +129,34 @@ public Command remove(EditingDomain ed, IBeanCategoryAssignment bca) {
 		return bcaHelper.getAllBeanCategories(sei, catBeanClazz);
 	}
 	
+	@Override
+	public List<BeanCategoryAssignment> getCategoryAssignments() {
+		ArrayList<BeanCategoryAssignment> beans = new ArrayList<BeanCategoryAssignment>();
+		
+		for (CategoryAssignment ca : sei.getCategoryAssignments()) {

```

Comment:
Isn't it possible to directly use the BeanCategoryAssignmentHelper by calling getAllBeanCategories(sei, ABeanCategoryAssignment.class) here?

Reply:
I don't see the advantage, because we use `BeanCategoryAssignment `here and not `ABeanCategoryAssignment`. Or do you want to change the contract to `ABeanCategoryAssignment`? If I remember correctly we use `BeanCategoryAssignment ` because there were some problems in the set method when using the abstract class instead.